x-pack/plugins/enterprise_search/server/lib/ml/start_ml_model_download.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { MlPutTrainedModelRequest } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { ElasticsearchClient } from '@kbn/core/server';
+import { MlTrainedModels } from '@kbn/ml-plugin/server';
+
+import { MlModelDeploymentState, MlModelDeploymentStatus } from '../../../common/types/ml';
+
+import { getMlModelDeploymentStatus } from './get_ml_model_deployment_status';
+import {
+  isNotFoundExceptionError,
+  throwIfNotAcceptableModelName,
+} from './ml_model_deployment_common';
+
+export const startMlModelDownload = async (
+  modelName: string,
+  esClient: ElasticsearchClient | undefined,
+  trainedModelsProvider: MlTrainedModels | undefined
+): Promise<MlModelDeploymentStatus> => {
+  if (!trainedModelsProvider || !esClient || !esClient.ml) {
+    throw new Error('Machine Learning is not enabled');
+  }
+
+  // before anything else, check our model name
+  // to ensure we only allow those names we want
+  throwIfNotAcceptableModelName(modelName);
+
+  try {
+    // try and get the deployment status of the model first
+    // and see if it's already deployed or deploying...
+    const deploymentStatus = await getMlModelDeploymentStatus(modelName, trainedModelsProvider);
+    const deploymentState = deploymentStatus?.deploymentState || MlModelDeploymentState.NotDeployed;
+
+    // if we're downloading or already started / starting / done
+    // return the status
+    if (deploymentState !== MlModelDeploymentState.NotDeployed) {
+      return deploymentStatus;
+    }
+  } catch (error) {
+    // don't rethrow the not found here -
+    // if it's not found there's a good chance it's not started
+    // downloading yet
+    if (!isNotFoundExceptionError(error)) {
+      throw error;
+    }
+  }
+
+  // we're not downloaded yet - let's initiate that...
+  const putRequest: MlPutTrainedModelRequest = {
+    // @ts-expect-error @elastic-elasticsearch inference_config can be optional
+    body: {
+      input: {
+        field_names: ['text_field'],
+      },
+    },
+    model_id: modelName,
+  };
+
+  // this will also sync our saved objects for us
+  await esClient.ml.putTrainedModel(putRequest);
```

Comment:
ml endpoints can't be called directly from elasticsearch. you need to use the functions shared in `trainedModelsProvider` 
e.g. `trainedModelsProvider.putTrainedModel`

There is currently a [PR](https://github.com/elastic/kibana/pull/155822) open to correct the types, once this is merged you'll be able see `putTrainedModel`. but if you need to test before that, `putTrainedModel` does exist, it's just that the types say it doesn't.